apps/src/templates/sectionProgress/sectionProgressRedux.js

```diff
@@ -222,14 +222,35 @@ export default function sectionProgress(state=initialState, action) {
           course.script_ids.forEach(id => idMap[id] = true);
         }
       });
-
       validScripts = validScripts.filter(script => idMap[script.id]);
+
+      var scriptId;
+      switch (true) {
+        // When there is a scriptId already in state.
+        case !!state.scriptId:
+          scriptId = state.scriptId;
+          break;
+        // When there is an assigned course, set scriptId to the first script in the assigned course.
+        case !!action.assignedCourseId:
+          action.validCourses.forEach(course => {
```

Comment:
not necessary to do right now, but you can return early from your `forEach` loop once `scriptId` has been assigned a value